rpcs3/Emu/Cell/lv2/sys_rsx.cpp

```diff
@@ -60,11 +77,84 @@ s32 sys_rsx_memory_free(u32 mem_handle)
  * @param mem_ctx (IN): mem_ctx given by sys_rsx_memory_allocate
  * @param system_mode (IN):
  */
-s32 sys_rsx_context_allocate(vm::ptr<u32> context_id, vm::ptr<u32> lpar_dma_control, vm::ptr<u32> lpar_driver_info, vm::ptr<u32> lpar_reports, u64 mem_ctx, u64 system_mode)
+s32 sys_rsx_context_allocate(vm::ptr<u32> context_id, vm::ptr<u64> lpar_dma_control, vm::ptr<u64> lpar_driver_info, vm::ptr<u64> lpar_reports, u64 mem_ctx, u64 system_mode)
 {
-	sys_rsx.todo("sys_rsx_context_allocate(context_id=*0x%x, lpar_dma_control=*0x%x, lpar_driver_info=*0x%x, lpar_reports=*0x%x, mem_ctx=0x%llx, system_mode=0x%llx)",
+	sys_rsx.warning("sys_rsx_context_allocate(context_id=*0x%x, lpar_dma_control=*0x%x, lpar_driver_info=*0x%x, lpar_reports=*0x%x, mem_ctx=0x%llx, system_mode=0x%llx)",
 		context_id, lpar_dma_control, lpar_driver_info, lpar_reports, mem_ctx, system_mode);
 
+	auto m_sysrsx = fxm::make<SysRsxConfig>();
+	if (!m_sysrsx)
+		fmt::throw_exception("sys_rsx_context_allocate called twice.");
+
+	vm::falloc(0x40000000, 0x10000000, vm::rsx_context);
```

Comment:
On Windows, too, its saved in the VAD.
Anyway, I'd agree with you on other types of applications, but as this is virtual memory on virtual memory, I believe it'll faster than checking if you need to alloc more in realtime. It isn't wasted memory, too, because it isn't used and no one will access it, so it shouldn't page fault.